src/Tribe/Main.php

```diff
@@ -3919,17 +3919,113 @@ public function isOrganizer( $postId = null ) {
 		}
 
 		/**
-		 * Get a "previous/next post" link for events. Ordered by start date instead of ID.
+		 * Modify the WHERE clause of query when fetching next/prev posts so events with identical times are not excluded
+		 *
+		 * This method ensures that when viewing single events that occur at a given time, other events
+		 * that occur at the exact same time are are not excluded from the prev/next links
+		 *
+		 * @since 4.0.2
+		 *
+		 * @param string $where_sql WHERE SQL statement
+		 * @param WP_Query $query WP_Query object
+		 *
+		 * return string
+		 */
+		public function get_closest_event_where( $where_sql ) {
+			// if we are in this method, we KNOW there is a section of the SQL that looks like this:
+			//     ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) [<|>] '2015-01-01 00:00:00' )
+			// What we want to do is to extract all the portions of the WHERE BEFORE that section, all the
+			// portions AFTER that section, and then rebuild that section to be flexible enough to include
+			// events that have the SAME datetime as the event we're comparing against.  Sadly, this requires
+			// some regex-fu.
+			//
+			// The end-game is to change the known SQL line (from above) into the following:
+			//
+			//  (
+			//    ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) [<|>] '2015-01-01 00:00:00' )
+			//    OR (
+			//      ( table.meta_key = '_EventStartDate' AND CAST( table.meta_value AS DATETIME ) = '2015-01-01 00:00:00' )
+			//      AND
+			//      table.post_id [<|>] POST_ID
+			//    )
+			//  )
```

Comment:
When the query is executed, the `ORDER BY` is set to sort by `EventStartDate` and then `post_id` (ASC/DESC depending on the state of next/prev), which causes ID 1 to be fetched in the last example.